version: "3"
services:
  pihole:
    image: pihole/pihole:v4.0_armhf
    ports:
      - "53:53/tcp"
      - "53:53/udp"
      - "80:80/tcp"
      - "443:443/tcp"
    volumes:
      - pihole_data:/etc/pihole
      - dnsmasq_data:/etc/dnsmasq.d
    environment:
      ServerIP: 192.168.2.2
      # jwilder/proxy envs, see readme for more info
      PROXY_LOCATION: pihole
      VIRTUAL_HOST: pihole.local
      VIRTUAL_PORT: 80
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        - condition: on-failure
volumes:
  pihole_data:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /var/rpi/pihole/etc/pihole
  dnsmasq_data:
    driver: local
    driver_opts:
      o: bind
      type: none
      device: /var/rpi/pihole/etc/dnsmasq.d
  # WARNING: if this log don't exist as a file on the host already
  # docker will try to create a directory in it's place making for lots of errors
  - '/var/log/pihole.log:/var/log/pihole.log'
